go-controller/pkg/config/config.go

```diff
@@ -854,7 +878,7 @@ func buildGatewayConfig(ctx *cli.Context, cli, file *config) error {
 		}
 	}
 	// And CLI overrides over config file and default values
-	if err := overrideFields(&Gateway, &cli.Gateway); err != nil {
+	if err := overrideFields(&Gateway, &cli.Gateway, nil); err != nil {
```

Comment:
some config savedGateway is just empty values and no real defaults, so I just pass nil for those. Maybe I should pass var instead in case in the future there is a default value assigned for these...